### ISSUE TITLE
Option to skip downloading if the file already exists

### ISSUE THREAD

**Issue by aminya** [kevva/download#211]

**Comment by naomi443** [comment]
Great idea

**Comment by Edim7** [comment]
I think such feature might just work better implemented on your own. A simple way would be to use fs.existsSync( path + filename) to check if the file already exists presuming your name has the version of the file in question.

**Comment by naomi443** [comment]
Sorry hon, I still have no idea what you are talking about?

**Comment by aminya** [comment]
Ideally, I want an option that optionally can accept a function used for version checking.

Passing a boolean:
```ts
download(url, destination, { skipReDownload: true})
```

Passing a function that returns a boolean:
```ts
download(url, destination, { skipReDownload: versionCheckFunction})

function versionCheckFunction(path_to_file: string) {
	// do any version check that you want to do here and return a boolean

	return true // skip redownload
	return false // the file is old -> redownload
}
```

**Comment by naomi443** [comment]
You may aswell be talking Double Dutch to me. Not a developer. 🙄🙄🤔🤪

**Comment by Edim7** [comment]
> Ideally, I want an option that optionally can accept a function used for version checking.
> 
> Passing a boolean:
> 
> ```ts
> download(url, destination, { skipReDownload: true})
> ```
> 
> Passing a function that returns a boolean:
> 
> ```ts
> download(url, destination, { skipReDownload: versionCheckFunction})
> 
> function versionCheckFunction(path_to_file: string) {
> 	// do any version check that you want to do here and return a boolean
> 
> 	return true // skip redownload
> 	return false // the file is old -> redownload
> }
> ```

Yeah that was my point, if you were to create your own version check function what stops you from doing an if statement and checking for the status of the custom function, and then decide if you are going to run the download or not? Based on that if you can provide output that will signify that the exact same version already exists locally.

**Comment by aminya** [comment]
The version check is optional and only for those who need extra flexibility (which you should always consider when designing an interface). For my personal use, I am happy to just pass `true` to `skipReDownload`.
